### PR TITLE
Added id parameter and readme changes

### PR DESCRIPTION
**Pull request by tobenxe**
Hey!

Just added the ability to increase the rate limit on the api by generating a token that can be sent with the request. The token can be gotten from the main site (triptocarbon.xyz). 

I quickly edited the main file to reflect the change. I can add tests for the new parameter if you need.

### CODE REVIEW

File: trip-to-carbon.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
-import axios from 'axios';
+import axios from 'axios'
```
Comment (EvanHahn) [inline trip-to-carbon.ts]: Why did this line, and many others in this file, change?

File: trip-to-carbon.ts [hunk 2]
```diff
@@ -23,10 +24,12 @@ type CarbonFootprintOptions = CarbonFootprintBaseOptions & (
   CarbonFootprintFuelOptions
 )
 
-export async function carbonFootprint(options: CarbonFootprintOptions): Promise<number> {
+export async function carbonFootprint (options: CarbonFootprintOptions): Promise<number> {
   const baseUrl = 'baseUrl' in options ? options.baseUrl : 'https://api.triptocarbon.xyz'
-
   const params = new URLSearchParams()
+
+  if ('id' in options) params.set('id', options.id!)
```
Comment (EvanHahn) [inline trip-to-carbon.ts]: Why does `options.id!` need a `!` at the end? Could this be `params.set('id', options.id)`?

File: README.md [hunk 3]
```diff
@@ -32,4 +32,26 @@ const fuelFootprint = await carbonFootprint({
...
+  id: 'SomeIdFromGottenTheSite',
+  distance: {
+    amount: 10,
+    unit: 'miles',
+    mode: 'taxi'
+  }
+})
+console.log(`Your 10-mile taxi trip had a carbon footprint of ${taxiFootprint} kilograms.`)
+
+const fuelFootprint = await carbonFootprint({
```
Comment (EvanHahn) [inline README.md]: I don't think this second example is necessary...what do you think?

**Review by EvanHahn — CHANGES_REQUESTED** [review]
Perhaps we could pick a better name—something like `userId`? I think it might be unclear what "ID" means in this context.

File: trip-to-carbon.ts (hunk 2 above)
Comment (TobeNicol) [reply trip-to-carbon.ts]: Ah alright, I've never actually used typescript. I'll change it.

File: trip-to-carbon.ts (hunk 1 above)
Comment (tobenxe) [reply trip-to-carbon.ts]: I'm not actually sure, I have fixed the weird spacing issues.

**Comment by EvanHahn** [comment]
Thanks for updating! I'll take a look soon (probably sometime in the next 2 weeks).

File: README.md [hunk 4]
```diff
@@ -26,10 +27,12 @@ console.log(`Your 10-mile taxi trip had a carbon footprint of ${taxiFootprint} k
...
 const fuelFootprint = await carbonFootprint({
   country: 'GBR',
+  token: 'YourAppToken',  //remove this property if you do not have a token
   fuel: {
     amount: 456,
     unit: 'gallons',
     type: 'jetFuel'
   }
 })
+
```
Comment (EvanHahn) [inline README.md]: Is this new line necessary?

File: README.md [hunk 5]
```diff
@@ -16,6 +16,7 @@ import { carbonFootprint } from 'trip-to-carbon-xyz'
 
 const taxiFootprint = await carbonFootprint({
   country: 'USA',
+  token: 'YourAppToken',  //remove this property if you do not have a token
```
Comment (EvanHahn) [inline README.md]: Would you mind:

1. moving this to the top (as the first option)
1. changing the comment to just "optional"

File: README.md [hunk 6]
```diff
@@ -26,10 +27,12 @@ console.log(`Your 10-mile taxi trip had a carbon footprint of ${taxiFootprint} k
 
 const fuelFootprint = await carbonFootprint({
   country: 'GBR',
+  token: 'YourAppToken',  //remove this property if you do not have a token
```
Comment (EvanHahn) [inline README.md]: Same comment from above applies here, too.

File: trip-to-carbon.ts [hunk 7]
```diff
@@ -71,4 +74,4 @@ function transformCountryOption(countryOption: string) {
     default:
       return 'def'
   }
-}
\ No newline at end of file
+}
```
Comment (EvanHahn) [inline trip-to-carbon.ts]: Why did this change?

File: trip-to-carbon.ts [hunk 8]
```diff
@@ -27,6 +28,8 @@ export async function carbonFootprint(options: CarbonFootprintOptions): Promise<
   const baseUrl = 'baseUrl' in options ? options.baseUrl : 'https://api.triptocarbon.xyz'
 
   const params = new URLSearchParams()
+
+  if('token' in options) params.set('appTkn', options.token!)
```
Comment (EvanHahn) [inline trip-to-carbon.ts]: Could you format this differently?

```ts
if ('token' in options) {
  params.set('appTkn', options.token!)
}
```

**Review by EvanHahn — CHANGES_REQUESTED** [review]

File: trip-to-carbon.ts (hunk 7 above)
Comment (tobenxe) [reply trip-to-carbon.ts]: Not sure why that changed, the only difference I could spot was a new line in my commit at the end of the file.

**Comment by EvanHahn** [comment]
I merged this manually in 13061d3b7fa9aa9314dd32dd9d1ba1ae03a020a7 and deployed it in version 0.2.0.

Thanks!

**Pull request closed without merging**